website/server/libs/webhook.js

```diff
@@ -1,31 +1,122 @@
-import { each } from 'lodash';
 import { post } from 'request';
 import { isURL } from 'validator';
 import logger from './logger';
 
-let _sendWebhook = (url, body) => {
+function sendWebhook (url, body) {
   post({
     url,
     body,
     json: true,
-  }, (err) => logger.error(err));
-};
-
-let _isInvalidWebhook = (hook) => {
-  return !hook.enabled || !isURL(hook.url);
-};
-
-export function sendTaskWebhook (webhooks, data) {
-  each(webhooks, (hook) => {
-    if (_isInvalidWebhook(hook)) return;
-
-    let body = {
-      direction: data.task.direction,
-      task: data.task.details,
-      delta: data.task.delta,
-      user: data.user,
-    };
-
-    _sendWebhook(hook.url, body);
+  }, (err) => {
+    if (err) {
+      logger.error(err);
+    }
   });
 }
+
+function isValidWebhook (hook) {
+  return hook.enabled && isURL(hook.url);
+}
+
+export class WebhookSender {
+  constructor (options = {}) {
+    this.type = options.type;
+    this.transformData = options.transformData || WebhookSender.defaultTransformData;
+    this.webhookFilter = options.webhookFilter || WebhookSender.defaultWebhookFilter;
+  }
+
+  static defaultTransformData (data) {
+    return data;
+  }
+
+  static defaultWebhookFilter () {
+    return true;
+  }
+
+  send (webhooks, data) {
+    let hooks = Object.keys(webhooks)
+      .map(hook => webhooks[hook])
+      .filter(isValidWebhook)
```

Comment:
all these `filter` are nice but it's basically iterating 3 times over an array. I'd combine the three or use lodash chaining